src/pytom_tm/extract.py

```diff
@@ -247,6 +247,12 @@ def extract_particles(
             for origin, size in zip(job.search_origin, job.search_size)
         ]
         tomogram_mask = tomogram_mask[*slices]
+
+        # If tomogram_mask is stored as signed bytes we need to convert it to unsigned:
+        if tomogram_mask.dtype == "int8":
+            if np.any(tomogram_mask < 0):
+                tomogram_mask = (tomogram_mask > -128).astype("int8")
```

Comment:
I would join these statements and make them strict to only trigger on misread signed binary masks
```suggestion
        if (tomogram_mask.dtype == "int8" and 
            -128 <= tomogram_mask.min() and tomogram_mask.max() <= -127
            ):
            tomogram_mask = (tomogram_mask > -128).astype("int8")
```